framework/replay_buffer.py

```diff
@@ -84,6 +99,27 @@ def sample_trajectories(self, env: Optional[VecNormalize] = None):
 
         return first_trajectory, second_trajectory
 
+
+    def _get_samples(self, batch_inds: np.ndarray, env: Optional[VecNormalize] = None) -> ReplayBufferSamples:
+        # Sample randomly the env idx
+        env_indices = np.random.randint(0, high=self.n_envs, size=(len(batch_inds),))
+
+        if self.optimize_memory_usage:
+            next_obs = self._normalize_obs(self.observations[(batch_inds + 1) % self.buffer_size, env_indices, :], env)
+        else:
+            next_obs = self._normalize_obs(self.next_observations[batch_inds, env_indices, :], env)
+
+        data = (
+            self._normalize_obs(self.observations[batch_inds, env_indices, :], env),
+            self.actions[batch_inds, env_indices, :],
+            next_obs,
+            # Only use dones that are not due to timeouts
+            # deactivated by default (timeouts is initialized as an array of False)
+            (self.dones[batch_inds, env_indices] * (1 - self.timeouts[batch_inds, env_indices])).reshape(-1, 1),
+            self._normalize_reward(self.ground_truth_rewards[batch_inds, env_indices].reshape(-1, 1), env),
+        )
+        return ReplayBufferSamples(*tuple(map(self.to_torch, data)))
```

Comment:
We should consider replacing ReplayBufferSamples() with a custom implementation, such as CustomReplayBufferSamples(NamedTuple), because ReplayBufferSamples() limits us to six positional arguments, which does not meet our requirement for seven arguments. [This is how I suggest we could implement it](https://github.com/jualat/CleanRLHF/tree/18-relabeling-added-CustomReplayBufferSamples)

Reply:
Yes, you're right, implementation should follow in some minutes :+1: